imap/index.c

```diff
@@ -5538,50 +5429,50 @@ static int extract_attachment(const char *type, const char *subtype,
         if (r) {
             syslog(LOG_ERR,
                    "extract_attachment: failed to read response for GET %s/%s",
-                   ctx->path, guidstr);
-            r = IMAP_IOERROR;
-            goto done;
+                   ext->path, guidstr);
+            statuscode = 599;
```

Comment:
Rather than bailing on pretty much any error, we create a synthetic failure code!